File: Alkitab/src/main/java/yuku/alkitab/base/util/AppLog.java
package yuku.alkitab.base.util;

import android.util.Log;
import com.crashlytics.android.Crashlytics;

public class AppLog {
	public static void d(String tag, String msg) {
		Crashlytics.log(Log.DEBUG, tag, msg);
	}

	public static void d(String tag, String msg, Throwable ex) {
		Crashlytics.log(Log.DEBUG, tag, msg);
		Crashlytics.logException(ex);
		Log.d(tag, msg, ex);
	}

	public static void i(String tag, String msg) {
		Crashlytics.log(Log.INFO, tag, msg);
	}

	public static void i(String tag, String msg, Throwable ex) {
		Crashlytics.log(Log.INFO, tag, msg);
		Crashlytics.logException(ex);
		Log.i(tag, msg, ex);
	}

	public static void w(String tag, String msg) {
		Crashlytics.log(Log.WARN, tag, msg);
	}

	public static void w(String tag, String msg, Throwable ex) {
		Crashlytics.log(Log.WARN, tag, msg);
		Crashlytics.logException(ex);
		Log.w(tag, msg, ex);
	}

	public static void e(String tag, String msg) {
		Crashlytics.log(Log.ERROR, tag, msg);
	}

	public static void e(String tag, String msg, Throwable ex) {
		Crashlytics.log(Log.ERROR, tag, msg);
		Crashlytics.logException(ex);
		Log.e(tag, msg, ex);
	}
}

File: Alkitab/src/main/java/yuku/alkitab/base/util/AppLog.kt
package yuku.alkitab.base.util

import android.util.Log
import com.google.firebase.crashlytics.FirebaseCrashlytics

object AppLog {
    private val crashlytics = FirebaseCrashlytics.getInstance()

    @JvmStatic
    @JvmOverloads
    fun d(tag: String?, msg: String?, ex: Throwable? = null) {
        val t = tag ?: "NO-TAG"
        val m = msg ?: "null"
        if (ex == null) {
            Log.d(t, m)
        } else {
            Log.d(t, m, ex)
        }
        crashlytics.log("D/$t: $m")
        if (ex != null) crashlytics.recordException(ex)
    }

    @JvmStatic
    @JvmOverloads
    fun i(tag: String?, msg: String?, ex: Throwable? = null) {
        val t = tag ?: "NO-TAG"
        val m = msg ?: "null"
        if (ex == null) {
            Log.i(t, m)
        } else {
            Log.i(t, m, ex)
        }
        crashlytics.log("I/$t: $m")
        if (ex != null) crashlytics.recordException(ex)
    }

    @JvmStatic
    @JvmOverloads
    fun w(tag: String?, msg: String?, ex: Throwable? = null) {
        val t = tag ?: "NO-TAG"
        val m = msg ?: "null"
        if (ex == null) {
            Log.w(t, m)
        } else {
            Log.w(t, m, ex)
        }
        crashlytics.log("W/$t: $m")
        if (ex != null) crashlytics.recordException(ex)
    }

    @JvmStatic
    @JvmOverloads
    fun e(tag: String?, msg: String?, ex: Throwable? = null) {
        val t = tag ?: "NO-TAG"
        val m = msg ?: "null"
        if (ex == null) {
            Log.e(t, m)
        } else {
            Log.e(t, m, ex)
        }
        crashlytics.log("E/$t: $m")
        if (ex != null) crashlytics.recordException(ex)
    }
}
